tests/convert_matrix_typed.c

```diff
@@ -0,0 +1,43 @@
+#include "bml.h"
+#include "../typed.h"
+
+#include <complex.h>
+#include <math.h>
+#include <stdlib.h>
+
+int TYPED_FUNC(
+    test_import_export) (
+    const int N,
+    const bml_matrix_type_t matrix_type,
+    const bml_matrix_precision_t matrix_precision,
+    const int M)
+{
+    bml_matrix_t *A;
+    REAL_T *A_dense;
+    REAL_T *B_dense;
+
+    A_dense = bml_allocate_memory(sizeof(REAL_T) * N * N);
+    for (int i = 0; i < N * N; i++)
+    {
+        A_dense[i] = rand() / (double) RAND_MAX;
+    }
+    A = bml_import_from_dense(matrix_type, matrix_precision, dense_row_major,
```

Comment:
@cnegre How should we test the `bml_convert` function? We could generate a dense matrix, convert that to bml, and then convert that bml to another bml. What do you think?

The other question we should think about is the following: Since `bml_convert` can convert from any bml matrix to any bml matrix, we should really run `(N_t N_p)^2` tests, where `N_t` is the number of matrix types and `N_p` the number of matrix precisions. Currently we can't do that automatically through CMake. Do we also want to add some mechanism in CMake to allow for this double testing? What do you think?